contributors/devel/security-release-process.md

```diff
@@ -82,14 +115,54 @@ The communication to users should be actionable. They should know when to block
 
 **Fix Release Day** (Completed within 1-21 days of Disclosure)
 
-- The Release Managers will ensure all the binaries are built, publicly available, and functional before the Release Date.
-  - TODO: this will require a private security build process.
-- The Release Managers will create a new patch release branch from the latest patch release tag + the fix from the security branch. As a practical example if v1.5.3 is the latest patch release in kubernetes.git a new branch will be created called v1.5.4 which includes only patches required to fix the issue.
-- The Fix Lead will cherry-pick the patches onto the master branch and all relevant release branches. The Fix Team will LGTM and merge.
-- The Release Managers will merge these PRs as quickly as possible. Changes shouldn't be made to the commits even for a typo in the CHANGELOG as this will change the git sha of the already built and commits leading to confusion and potentially conflicts as the fix is cherry-picked around branches.
+- Before the Release Date, the Release Manager(s) will ensure all the binaries
+  are built, publicly available, and functional.
+  - Note: since we ship source with our binary releases, as soon as binary
+    artifacts are publicly available, the fixes (and vulnerabilities) are
+    effectively public.
+  - CI testing of the private patch release branch should provide confidence in
+    the release.
+  - The Release Manager will use release tooling (i.e. `anago`) to
+    - Compute the version for the new security patch release; in the previous
+      example, the `release-2.4.7` branch would produce a `v2.4.8` release.
+    - Verify that no public patch release has occurred since the security branch
+      was cut (in this example, make sure that `v2.4.8` has not been tagged
+      publicly).
+    - Perform a normal release build from the private release branch, staging
+      binary artifacts locally (or possibly in a private bucket).
+    - Push the new patch release tag to the private repo.
+    - The tooling should **not** upload any artifacts to public storage buckets or
+      create a draft release on the public GitHub repo.
+- On the Release Date, the Release Manager will use tooling to
+  - Verify that nothing new has been pushed to the private release repository,
+    and that no new public patch release has occurred.
+      - If there are new commits on the private release branch, or if a new
+        public release has occurred, the previously build artifacts will be
+        discarded.
+  - Publish the release.
+    - Upload prebuilt binary artifacts (from previous step) to GCS, gcr.io, and
+      GitHub.
+    - Update the `CHANGELOG.md` in the **public** `master` branch
+    - Create a new release on the **public** kubernetes GitHub
+    - Send email announcing the release
+  - Merge the private patch release branch into the **public** release branch
```

Comment:
> including both the commits and release tag

the release tag would already have been pushed exactly, not merged, as part of creating the v2.4.8 release

I suppose the commit that bumps the version to v2.4.8 could be merged, though it's misleading if someone checks out that commit and builds it

> Advance the public release branch to the next beta.0 release, by updating tags and metadata as necessary.

If anago creates the bump commit to `v2.4.9-beta.0` as part of the `v2.4.8` release, do we get the commit for free by merging in the hotfix branch? All we need to do then is tag the result of that merge `v2.4.9-beta.0`, right?